src/components/outline-nav/components/outline-link/index.tsx

```diff
@@ -0,0 +1,21 @@
+import classNames from 'classnames'
+import type { OutlineLinkProps } from '../../types'
+import s from './outline-link.module.css'
+
+/**
+ * Renders an individual outline nav link.
+ */
+function OutlineLink({ title, url, isActive }: OutlineLinkProps) {
+	return (
+		<a
+			className={classNames(s.root, { [s.isActive]: isActive })}
+			href={url}
```

Comment:
💭 Thought: we could update the props interface here to accept `slug` rather than `url`, and then do `href={`#${slug}`}. This would add some opinion to the component (ie it'd no longer be able to render arbitrary URLs), but could simplify other parts of the code here (eg `get-item-slugs` would be much simpler, would work more smoothly with `isActiveSection`).

I'm a little on the fence about which direction to go here, I initially leaned towards baking in as little opinion as possible, so that's what I've implemented so far. Open to thoughts 🙇 cc @BrandonRomano

Reply:
It's taking me a while to type this out because I myself am conflicted.

On one hand, I think that the lack of opinion is the right move here.  Right now we have plans to use the OutlineNav _only_ in the sidecar, but I could imagine at some point in the future this component could be leveraged and may have a different set of business rules. We're really only going to interface with the OutlineNav from a few higher level implementations, so the interface doesn't need to really be optimized (and we can provide the _slug_ as the interface at one of those higher order implementations and do the mapping).

On the other hand, I think that it's probably unlikely that we ever use this OutlineNav anywhere else than where we plan, so I could see the argument for the other way and implementing the business logic of in-page-links only here.

I think in the spirit of this being a view-model & agnostic to business logic, we probably go the way you have implemented.

Reply:
Sweet - yeah I was kind of thinking the same thing, maybe the issue here is more that the business logic and higher level components could use a little refinement. I'll leave as-is 👍

Reply:
Agreed, I'd lean towards keeping this component unaware of the rendering context and leaving it as `url`.

Reply:
💭 I'd also suggest keep this component as-is!